common/app/views/support/cleaner/AmpEmbedCleaner.scala

```diff
@@ -60,6 +60,24 @@ case class AmpEmbedCleaner(article: Article) extends HtmlCleaner {
 
   }
 
+  def cleanAmpSoundcloud(document: Document) = {
+    document.getElementsByClass("element-audio").filter { audioElement: Element =>
+      audioElement.getElementsByTag("iframe").length != 0
+    }.foreach { audioElementWithIframe: Element =>
+      audioElementWithIframe.getElementsByTag("iframe").foreach { iframeElement: Element =>
+        val soundcloudUrl = iframeElement.attr("src")
+        val pattern = "%2Ftracks%2F(\\d+)".r
+        val trackId = pattern.findAllIn(soundcloudUrl).matchData.map { matches => matches.group(1) }.mkString
```

Comment:
do you really want to find them all, and then concatenate them together? or should you be finding the first, and only create the element if there's a match?